src/utils/utils.ts

```diff
@@ -1117,3 +1118,51 @@ export const validConcentrationMidPriceTick = (
 
   return midPriceTick
 }
+
+type LiquidityTickWithoutIndex = Omit<LiquidityTick, 'index'>
+
+export const getLiquidityTicksByPositionsList = (
```

Comment:
You probably ignore the comment with using `filter` function, instead of skipping some loop iterations. I mean something like this:
```Typescript
export const getLiquidityTicksByPositionsList = (
  poolKey: PoolKey,
  positions: Position[]
): LiquidityTick[] => {
  const ticks: Record<number, LiquidityTickWithoutIndex> = {};

  const isMatchingPoolKey = (position: Position) => 
    poolKeyToString(position.poolKey) === poolKeyToString(poolKey);

  const updateTicks = (position: Position) => {
    updateTick(ticks, Number(position.lowerTickIndex), position.liquidity, true);
    updateTick(ticks, Number(position.upperTickIndex), position.liquidity, false);
  };

  positions
    .filter(isMatchingPoolKey)
    .forEach(updateTicks);

  return Object.entries(ticks).map(([index, { liquidityChange, sign }]) => ({
    index: BigInt(index),
    liquidityChange,
    sign
  }));
};
```

Reply:
but in worst case scenario it would require full second loop, and in my opinion it doesn't help much with readability

Reply:
okay, we can keep the old code and it's not a big difference. Regarding readability, I disagree. it's much more readable. I always look at readability from the perspective of someone who doesn't know the codebase. Some variable names guide the intention step by step on what needs to be done